src/Components/ActivationKeysTable/ActivationKeysTable.js

```diff
@@ -14,9 +14,62 @@ const ActivationKeysTable = (props) => {
     role: 'Role',
     serviceLevel: 'SLA',
     usage: 'Usage',
+    updatedAt: 'Updated Date',
   };
   const { isLoading, error, data } = useActivationKeys();
+  console.log(data);
+  const [activeSortIndex, setActiveSortIndex] = React.useState(null);
+  const [sortedData, setSortedData] = React.useState([]);
+  const [activeSortDirection, setActiveSortDirection] = React.useState(null);
   const location = useLocation();
+  React.useEffect(() => {
+    if (data) {
+      setSortedData(data);
+    }
+  }, [data]);
+
+  const getSortParams = (columnIndex) => ({
+    sortBy: {
+      index: activeSortIndex,
+      direction: activeSortDirection,
+      defaultDirection: 'asc',
+    },
+    onSort: (_event, index, direction) => {
+      setActiveSortIndex(index);
+      setActiveSortDirection(direction);
+      const sorted = [...sortedData].sort((a, b) => {
+        const aValue = a[columnNames[index]];
+        const bValue = b[columnNames[index]];
+        if (index === 4) {
+          const aDate = new Date(a.updatedAt);
+          const bDate = new Date(b.updatedAt);
+          if (direction === 'asc') {
+            return aDate - bDate;
+          } else {
+            return bDate - aDate;
+          }
+        }
+        if (aValue < bValue) return direction === 'asc' ? -1 : 1;
+        if (aValue > bValue) return direction === 'asc' ? 1 : -1;
+        return 0;
+      });
+      setSortedData(sorted);
+    },
+    columnIndex,
+  });
+
+  const printDate = (dateString) => {
+    const date = new Date(dateString);
+    if (isNaN(date)) {
+      return 'Invalid Date';
+    }
+    const month = date.getUTCMonth() + 1;
+    const day = date.getUTCDate();
+
+    return `${date.getUTCFullYear()}-${month < 10 ? `0${month}` : month}-${
+      day < 10 ? `0${day}` : day
+    }`;
+  };
```

Comment:
Polish: Same here, this could be a helper function. For both of these, if they're in a helper file we can easily write unit tests for this functionality too.